### ISSUE TITLE
Bug in Agenda Mode, forget to set second and millisecond at EventBase.prototype.splitEventByDay function

### ISSUE THREAD

**Issue by percy507** [syncfusion/ej2-react-ui-components#31]
I create an event at Month mode. 
And i want to show the event at MonthAgenda mode.

But, something bad happened.
My web page stucked. Cause the memory leaked.

Then i test the data with Schedule component. Finally i found the answer.

```js
// my test data, cause memory leaked
[{
     Id: '595613f3-61d8-45cf-b834-ea7bab0e7a31',
    Title: 'Test event',
    StartTime: '2019-08-15T17:13:46.880+08:00',
    EndTime: '2019-08-23T17:13:46.880+08:00'
}]

// my another test data, it works
[{
     Id: '595613f3-61d8-45cf-b834-ea7bab0e7a31',
    Title: 'Test event',
    StartTime: '2019-08-15T17:13:00.000+08:00',
    EndTime: '2019-08-23T17:13:00.000+08:00'
}]
```

So i read the source code. 

File path: `node_modules/@syncfusion/ej2-schedule/src/schedule/event-renderer/event-base.js`.

And notice the code below.

```js
EventBase.prototype.splitEventByDay = function(event) {
  var eventFields = this.parent.eventFields;
  var data = [];
  var eventStartTime = event[eventFields.startTime];
  var eventEndTime = event[eventFields.endTime];
  var isDifferentDate =
    util.resetTime(new Date(eventStartTime.getTime())) <
    util.resetTime(new Date(eventEndTime.getTime()));
  if (isDifferentDate) {
    var start = new Date(eventStartTime.getTime());
    var end = util.addDays(
      util.resetTime(new Date(eventStartTime.getTime())),
      1
    );
    var endDate =
      eventEndTime.getHours() === 0 && eventEndTime.getMinutes() === 0
        ? eventEndTime
        : util.addDays(eventEndTime, 1);
    var index = 1;
    var eventLength = util.getDaysCount(
      eventStartTime.getTime(),
      endDate.getTime()
    );
    while (end <= eventEndTime) {
      var app = extend({}, event);
      app[eventFields.startTime] = start;
      app[eventFields.endTime] = end;
      app.data = { index: index, count: eventLength };
      app.Guid = this.generateGuid();
      app.isSpanned = true;
      data.push(app);
      start = end;
      if (
        new Date(start.getTime()).setHours(0, 0, 0, 0) ===
          new Date(eventEndTime.getTime()).setHours(0, 0, 0, 0) &&
        !(end.getTime() === eventEndTime.getTime())
      ) {

        // *******************************
        // the below code forget the set the second and millisecond
        // so if original time has second or millisecond, the while loop will be a dead loop
        // *******************************
        end = new Date(
          new Date(start.getTime()).setHours(
            eventEndTime.getHours(),
            eventEndTime.getMinutes()
          )
        );
      } else {
        end = util.addDays(util.resetTime(new Date(start.getTime())), 1);
      }
      index++;
    }
  } else {
    data.push(event);
  }
  return data;
};
```

**Comment by vinithadevi** [comment]
Dear Customer,

Syncfusion greetings.

Currently it is not possible to render custom range of dates in timeline schedular and for the same we have already logged the feature request which can be tracked from the following link. It will be implemented in any one of our upcoming main release.
https://www.syncfusion.com/feedback/5075/custom-view-support-on-scheduler-with-user-defined-date-range

Mean time we suggest you to use interval option of view to customize the TimelineMonth and for the same we have prepared the below sample.
https://stackblitz.com/edit/rwc6ag-ey9vv4?file=index.ts 

Also we could reproduce the reported issue of “scroll bar not moving to visualize the today” and logged the below bug report and fix will be included in our upcoming Volume 3 release which is expected to roll out by end of September.We would appreciate your valuable patience until then.
https://www.syncfusion.com/feedback/8542/scroll-not-work-properly-in-timelinemonth-view 

Regards,
M.Vinitha devi
